### ISSUE TITLE
twisted.application.internet pydoctor errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5244]
|<img alt="magmatt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| magmatt reported|
|-|-|
|Trac ID|trac#5244|
|Type|enhancement|
|Created|2011-08-30 15:15:04Z|
|Branch|https://github.com/twisted/twisted/tree/pydoctor-errors-5244|

pydoctor output:
```
twisted.application.internet.CooperatorService.stopService:316 invalid ref to Deferred
twisted.application.internet.CooperatorService.stopService:316 invalid ref to Deferred
twisted.application.internet.StreamServerEndpointService.stopService:379 invalid ref to Deferred
twisted.application.internet.StreamServerEndpointService:321 invalid ref to IStreamServerEndpoint
twisted.application.internet.StreamServerEndpointService:321 invalid ref to IStreamServerEndpoint
twisted.application.internet.TimerService.stopService:293 invalid ref to Deferred
twisted.application.internet.TimerService.stopService:293 invalid ref to Deferred
twisted.application.internet._AbstractClient._getConnection:176 invalid ref to IConnector
twisted.application.internet._AbstractClient.stopService:169 invalid ref to Deferred
twisted.application.internet._AbstractClient.stopService:169 invalid ref to Deferred
twisted.application.internet._AbstractServer._getPort:123 invalid ref to IListeningPort
twisted.application.internet._AbstractServer.stopService:113 invalid ref to Deferred
twisted.application.internet._AbstractServer.stopService:113 invalid ref to Deferred
twisted.application.internet:0 invalid ref to IStreamServerEndpoint
twisted.application.internet:0 invalid ref to twisted.internet.strports.service
```

Attachments:

* [twisted.application.internet.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/aec/aec0f891c3b27ce85e4bd174dc21e51b4708d925/2fa1abe2638b0ae75c621ffd60f9ac676c34b156.diff) (6228 bytes) - added by magmatt on 2011-08-30 15:15:18Z - Patch

<details><summary>Searchable metadata</summary>

```
trac-id__5244 5244
type__enhancement enhancement
reporter__magmatt magmatt
priority__normal normal
milestone__ 
branch__branches_pydoctor_errors_5244 branches/pydoctor-errors-5244
branch_author__magmatt__thijs magmatt, thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1314717304000000 1314717304000000
changetime__1315833010000000 1315833010000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to magmatt|
|-|-|

Thanks!

  1. In `twisted/application/internet.py`, it looks like `C{twisted.internet.interfaces.IListeningPort}` in the documentation for the `_port` attribute could/should use `L{}` instead.
  1. Likewise for `C{twisted.internet.interfaces.IConnector}` on the `_connector` docs.
  1. In `twisted/application/service.py`, there's some malformed epytext: _L{Deferred<defer.Deferred}_ - shows up about 6 times.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [32509]) Branching to 'pydoctor-errors-5244'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [32510]) apply twisted.application.internet.diff, refs [#5244](https://github.com/twisted/twisted/issues/5244)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra removed owner|
|-|-|

I addressed the review comments, doc build results are [here](http://buildbot.twistedmatrix.com/builders/documentation/builds/1514).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Thanks!  Needs a (misc) news fragment, otherwise perfect.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [32526]) News file, refs [#5244](https://github.com/twisted/twisted/issues/5244)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [32527]) Merge pydoctor-errors-5244: Fixed incorrect epytext markup in docstrings.

Author: magmatt, thijs
Reviewer: exarkun
Fixes: [#5244](https://github.com/twisted/twisted/issues/5244)
